src/render/posts/does-tdd-work.html.md

```diff
@@ -0,0 +1,228 @@
+* The project developed by using TDD might be easier. 
+
+The conclusion of that paper were less positive than Janzen's as they rightly concluded the results were too varied and sample sizes too small to draw any positive conclusions.
+
+Another [controlled experiment conducted in 2012](http://www.ipr.mdh.se/pdf_publications/2345.pdf) once again concluded that TDD is probably a good thing but that more evidence is needed.
+
+[Another study from 2016](http://people.brunel.ac.uk/~csstmms/FucciEtAl_ESEM2016.pdf) looked at the effects of TDD compared to Test Last Development (TLD).
+
+>In this paper we reported a replication of an experiment
+[13] in which TDD was compared to a test-last approach.
+
+This study seemed to lean towards a verdict that TDD doesn't improve things over TLD:
+
+>Given the limitations presented in Section 5, it appears that
+TDD does not improve, nor deteriorate the participants’
+performance with respect to an iterative development technique
+in which unit tests are written after production code
+ 
+If you look at how the tests were conducted though you can see an interesting aspect of it is how programmers were to use iterative development. There was a fairly tight loop between test and production code for both TDD and TLD. Perhaps a tight iterative loop is more important than the order of test code?
+
+In my opinion this paper didn't provide enough data to give a conclusion one way or the other. It certainly didn't prove TDD was better or worse than writing all your tests after writing all your production code for a decent sized project. Also this study was focused in academia which can make it harder to relate to industry. 
+ 
+The final study we will glance at is one I just recently came across called [A dissection of the test-driven development process: does it really matter to test-first or to test-last?](https://arxiv.org/pdf/1611.05994.pdf) which is also by Fucci et al. This study actually appears to reach a similar conclusion to the previous study except that in this case the programmers involved work in industry. Also the comparisons were with iterative test last (ITL) development (which I think is a better term than TLD for this) and TDD.
+
+Another interesting conclusion in that study is that shorter cycle times (time between production code and test), to a point, do appear to lead to better quality.
+
+## Some Conclusions
+
+Looking at all the data it is fairly easy to say the answer to the questions 'Does TDD Work?' is inconclusive. 
+
+If the question become more specific at least there are some answers. TDD as a process is definitely better than some alternatives, such as no test process.
+
+The evidence appears to be heavily stacked in favour of short iterative test cycles, very similar to that prescribed in TDD literature, being significantly better for code quality with minimal impact on productivity. Small iterations with small testable blocks of code does appear to lead to more maintainable code. 
```

Comment:
Either "A process of small iterations ... does", or "Small iterations ... do" (I'm not sure if the latter sounds right?)